lib/types.go

```diff
@@ -41,6 +42,18 @@ type Options struct {
 	Thresholds map[string][]*Threshold `json:"thresholds"`
 }
 
+type SourceData struct {
```

Comment:
Yeah. One thing I think shouldn't be too hard to add later is support for "URL lists" where you send in a bunch of URLs, separated by newlines. That could (?) mean the URL runner would be able to consume two kinds of source data types: TypeURL and TypeURLList (e.g.).  But OK, let's remove the Type for now then?